common/src/main/java/com/google/auto/common/SuperficialValidation.java

```diff
@@ -63,10 +64,13 @@ public static boolean validateElements(Iterable<? extends Element> elements) {
         }
 
         @Override public Boolean visitType(TypeElement e, Void p) {
+          TypeMirror superclass = e.getSuperclass();
           return isValidBaseElement(e)
               && validateElements(e.getTypeParameters())
               && validateTypes(e.getInterfaces())
-              && validateType(e.getSuperclass());
+              && validateType(superclass)
+              && e.getInterfaces().stream().map(MoreTypes::asElement).allMatch(SuperficialValidation::validateElement)
```

Comment:
If you want, `e -> validateElement(e)` is a little shorter than the method reference

Reply:
Would prefer to leave as is - `e` conflicts with the existing method parameter and the IDE produces a warning with the lambda.